allowances/test/signature.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers'
+import { expect } from 'chai'
+import {
+  AbiCoder,
+  BigNumberish,
+  keccak256,
+  solidityPacked,
+  toUtf8Bytes,
+  ZeroAddress,
+} from 'ethers'
+
+import setup from './test-helpers/setup'
+
+describe('signature', async () => {
+  it('Generates expected transfer hash', async () => {
+    const { allowanceModule, alice } = await loadFixture(setup)
+
+    const allowanceAddress = await allowanceModule.getAddress()
+
+    const transfer = {
+      safe: '0x0000000000000000000000000000000000000010',
+      token: '0x0000000000000000000000000000000000000020',
+      to: '0x0000000000000000000000000000000000000030',
+      amount: 12345,
+      paymentToken: ZeroAddress,
+      payment: 0,
+      nonce: 0,
+    }
+
+    const {
+      DOMAIN_SEPARATOR_TYPEHASH,
+      ALLOWANCE_TRANSFER_TYPEHASH,
+      transferHash,
+    } = calculateTransferHash(
+      allowanceAddress,
+      (await alice.provider.getNetwork()).chainId,
+      transfer
+    )
+
+    expect(await allowanceModule.DOMAIN_SEPARATOR_TYPEHASH()).to.equal(
+      DOMAIN_SEPARATOR_TYPEHASH
+    )
+
+    expect(await allowanceModule.ALLOWANCE_TRANSFER_TYPEHASH()).to.equal(
+      ALLOWANCE_TRANSFER_TYPEHASH
+    )
+
+    expect(
+      await allowanceModule.generateTransferHash(
+        transfer.safe,
+        transfer.token,
+        transfer.to,
+        transfer.amount,
+        transfer.paymentToken,
+        transfer.payment,
+        transfer.nonce
+      )
+    ).to.equal(transferHash)
+  })
+})
+
+function calculateTransferHash(
```

Comment:
You can use https://docs.ethers.org/v6/api/hashing/#TypedDataEncoder in this function to simplify it

Reply:
Ok, just addressed this. We no longer assert on the separators, only on the hash as a whole.